magento/devdocs#: Cloud. Set ADMIN_URL via magneto-cloud command-line tool

atwixfirster opened the pull request:
## Purpose of this pull request

Magento support [recommends](https://support.magento.com/hc/en-us/articles/115005084874-Change-Magento-Admin-URL-on-Cloud) to use `magento-cloud` command line tool to change the alias of magento back-end.

This pull request (PR) adds a link to an example how to set Admin URL via `magento-cloud` command line tool.

## Affected DevDocs pages

<!-- REQUIRED List the affected pages on devdocs.magento.com (URLs). Not needed for large numbers of files. -->

- https://devdocs.magento.com/guides/v2.2/cloud/env/environment-vars_magento.html
- https://devdocs.magento.com/guides/v2.3/cloud/env/environment-vars_magento.html


## Links to Magento source code

<!--  OPTIONAL - REMOVE THIS SECTION IF NOT USED. If this pull request references a file in a Magento codebase repository, add it here. -->

- [Magento Cloud support](https://support.magento.com/hc/en-us/articles/115005084874-Change-Magento-Admin-URL-on-Cloud)

<!--
If you are fixing a GitHub issue, note it using GitHub keyword format (https://help.github.com/en/articles/closing-issues-using-keywords#closing-an-issue-in-a-different-repository) to close the issue when this pull request is merged. Example: `Fixes #1234`

`master` is the default branch. Merged pull requests to `master` go live on the site automatically. Any requested changes to content on the `master` branch must be related to the released codebase. Any content related to future releases goes in the `develop` branch.

See Contribution guidelines (https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md) for more information.
-->

Thank you!

devops-devdocs commented:
An admin must run tests on this PR before it can be merged.

hguthrie commented on guides/v2.2/cloud/env/environment-vars_magento.md:
```diff
@@ -14,17 +14,17 @@ Use the ADMIN variables in the following table to override credentials for the M
 
 If you want to change the values after installation, connect to your environment using ssh and use the the Magento CLI [`admin:user` command]({{ page.baseurl }}/install-gde/install/cli/install-cli-subcommands-admin.html) to create or edit the Magento Admin user credentials.
```
This could use a scrubbing also.

atwixfirster replied:
Also **fixed**

Thank you, @hguthrie

hguthrie commented on guides/v2.2/cloud/env/environment-vars_magento.md:
```diff
@@ -14,17 +14,17 @@ Use the ADMIN variables in the following table to override credentials for the M
 `ADMIN_URL`—The relative URL to access the Magento Admin UI. The default URL is `/admin`. For security reasons, Magento recommends that you change the default to a unique, custom Admin URL that is not easy to guess.
 
-If you need to change the Admin URL after installation, connect to your environment using SSH and use the Magento CLI [`magento setup:config`]({{ page.baseurl }}/install-gde/install/cli/install-cli-subcommands-deployment.html) command to change the URL.
+If you need to change the Admin URL after installation, connect to your environment using SSH and use the [`magento-cloud variable:set`]({{ page.baseurl }}/cloud/env/set-variables.html) command-line tool to change the URL.
```
Do you think this still implies connecting to remote?

atwixfirster replied:
Golden eye, @hguthrie !

**fixed**

hguthrie commented:
@atwixfirster I am looking into the fact that some variables cannot be set in this way. I don't see where the Cloud docs explicitly calls these out and so you may have stumbled on an opportunity where we can bring clear attention to this. I'll let you know what I find out...

hguthrie commented:
You can use the Magento Cloud CLI or web UI to update those variables for the Magento installation. After the Magento installation, you can not update `user data` in the same way. You would have to use the `bin/magento admin:user` method.

atwixfirster commented:
> You can use the Magento Cloud CLI or web UI to update those variables for the Magento installation. After the Magento installation, you can not update `user data` in the same way. You would have to use the `bin/magento admin:user` method.

@hguthrie , I do not understand. Per  your comment I can not use 
```
magento-cloud variable:set ADMIN_URL magento_A8v10 -e master
```
to change ADMIN_URL after Magento installation?

Thank you!

hguthrie commented:
Yes, the validator checks that `ADMIN_EMAIL`, `ADMIN_USERNAME`, `ADMIN_FIRSTNAME`, `ADMIN_LASTNAME`, and `ADMIN_PASSWORD` are used only during installation.

atwixfirster commented:
> Yes, the validator checks that `ADMIN_EMAIL`, `ADMIN_USERNAME`, `ADMIN_FIRSTNAME`, `ADMIN_LASTNAME`, and `ADMIN_PASSWORD` are used only during installation.

I've reverted back my changes regarding your notice.

Could you please review a final version.

Thank you, @hguthrie !

hguthrie commented:
running tests

hguthrie commented:
running tests

hguthrie commented:
This keeps failing build:

```- internally linking to /guides/v2.3/cloud/env/set-variables.html, which does not exist
  *  _site/guides/v2.3/cloud/env/environment-vars_magento.html (line 305)
     <a href="/guides/v2.3/cloud/env/set-variables.html"></a>```

hguthrie commented on guides/v2.2/cloud/env/environment-vars_magento.md:
```diff
@@ -14,17 +14,17 @@ Use the ADMIN variables in the following table to override credentials for the M
 `ADMIN_URL`—The relative URL to access the Magento Admin UI. The default URL is `/admin`. For security reasons, Magento recommends that you change the default to a unique, custom Admin URL that is not easy to guess.
 
-If you need to change the Admin URL after installation, connect to your environment using SSH and use the Magento CLI [`magento setup:config`]({{ page.baseurl }}/install-gde/install/cli/install-cli-subcommands-deployment.html) command to change the URL.
+If you need to change the Admin URL after installation, use the [`magento-cloud variable:set`]({{ page.baseurl }}/cloud/env/set-variables.html) command-line tool to change the URL.
```
To fix the broken link, do you think this would be an appropriate link to use?
https://devdocs.magento.com/guides/v2.2/cloud/before/before-setup-env-2_clone.html#change-the-magento-admin-variables

atwixfirster replied:
> uld be an appropriate link to use?
> https://devdocs.magento.com/guides/v2.2/cloud/before/before-setup-env-2_clone.html#change-the-magento-admin-variables

nope

Our links are:
- https://devdocs.magento.com/guides/v2.2/cloud/env/set-variables.html
- https://devdocs.magento.com/guides/v2.3/cloud/env/set-variables.html

atwixfirster commented:
> This keeps failing build:
> 
> ```
>   *  _site/guides/v2.3/cloud/env/environment-vars_magento.html (line 305)
>      <a href="/guides/v2.3/cloud/env/set-variables.html"></a>```
> ```

I've fixed a content. Removed link and added an example from https://devdocs.magento.com/guides/v2.2/cloud/env/set-variables.html ;)

It will help to pass tests.

@hguthrie , could you please review?

Thank you!

hguthrie commented:
running tests

hguthrie commented:
Victory! 🏆

atwixfirster commented:
> Victory! 🏆

yep

Thanks, @hguthrie 🌸
